src/borg/crypto/_crypto_helpers.c

```diff
@@ -4,7 +4,7 @@
 #include <openssl/opensslv.h>
 #include <openssl/hmac.h>
 
-#if OPENSSL_VERSION_NUMBER < 0x10100000L
+#if OPENSSL_VERSION_NUMBER < 0x10100000L || defined(LIBRESSL_VERSION_NUMBER)
```

Comment:
hmm, you don't actually check the libressl version number.

guess that's because there is no openssl 1.1 compatible libressl yet (not even planned?).

as we plan to maybe require openssl >= 1.1 in the future to support AES-OCB and chacha20-poly1305, the question is whether it makes sense to add compatibility for libressl now if it would then break again.

can you tell where you see the benefits of supporting libressl?

Reply:
I'm not aware of any plans to make the libressl API compatible with the newer openssl versions. I guess that they will support AES-OCB and chacha20-poly1305 but I don't know about that.

The benefit of supporting libressl is that some distributions switched to it and others (e. g. Exherbo) allow the user to choose. If borg does not support libressl it cannot be used on these distributions...

Reply:
hmm, looks like chacha20-poly1305 is already there, but not AES-OCB.

guess you can always use the binary of borg, which comes with openssl (and everything else) included.

Reply:
Sure, that should work. But of course that won't be packaged by the distributions and the users have to get it themselves.

If this fix is an issue in the long term, ok. But right now it seems trivial...
And my assumption is that libressl will provide all the functionality soon.